### PR TITLE
states: refactor step state class, add migration state (CRAFT-471)

### PR DESCRIPTION
**Pull request by cmatsuoka**
Refactor step state and add a migration state superclass to be used
when migrating overlay files.

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

File: craft_parts/state_manager/states.py [hunk 1]
```diff
@@ -80,6 +99,17 @@ def remove(part: Part, step: Step) -> None:
         state_file.unlink()
 
 
-def state_file_path(part: Part, step: Step) -> Path:
-    """Return the path to the state file for the give part and step."""
+def step_state_path(part: Part, step: Step) -> Path:
```
Comment (cjp256) [inline craft_parts/state_manager/states.py:102]: jw if we should prefix these methods with get_ or similar, but your call :D

**Review by cjp256 — APPROVED** [review]
LGTM

File: craft_parts/state_manager/states.py (hunk 1 above)
Comment (sergiusens) [reply craft_parts/state_manager/states.py:102]: that would be nice even if the operation is cheap

Comment (cmatsuoka) [reply craft_parts/state_manager/states.py:102]: Renamed `step_state_path()` and `overlay_migration_state_path()` to `get_step_state_path()` and `get_overlay_migration_state_path()`.

**Review by facundobatista — APPROVED** [review]
Looks great.

**Pull request merged**